### PR TITLE
Continue mapping from initial location

### PR DESCRIPTION
**Pull request by SteveMacenski**

### CODE REVIEW

**Comment by SteveMacenski** [comment]
things to resolve
- [x] deserializing off of a active session
- [x] fix readme image link
- [x] put my money where my mouth is and on the fly find the laser orientation
- [x] option to start at first node or where you left off in tooling + plugin

**Comment by SteveMacenski** [comment]
tested -- working and fixed an issue with solver reset state

**Comment by SteveMacenski** [comment]
@ivonaj ping :small_airplane:

File: lib/karto_sdk/src/Mapper.cpp [hunk 1]
```diff
@@ -2303,9 +2317,165 @@ namespace karto
...
+      // correct scan (if not first scan)
+      if (m_pUseScanMatching->GetValue() && pLastScan != NULL)
+      {
+        Pose2 bestPose;
+        m_pSequentialScanMatcher->MatchScan(pScan,
+            m_pMapperSensorManager->GetRunningScans(pScan->GetSensorName()),
+            bestPose,
+            covariance);
+        pScan->SetSensorPose(bestPose);
+      }
```
Comment (ivonaj) [inline lib/karto_sdk/src/Mapper.cpp:2373]: Lookin good! We got ourselves a new Process function

File: solvers/ceres_solver.cpp [hunk 2]
```diff
@@ -228,6 +228,32 @@ void CeresSolver::Clear()
...
+    delete nodes_;
+  }
+
+  nodes_ = new std::unordered_map<int, Eigen::Vector3d>();
+  problem_ = new ceres::Problem();
+  first_node_ = nodes_->end();
+
+  angle_local_parameterization_ = AngleLocalParameterization::Create();
+}
+
```
Comment (ivonaj) [inline solvers/ceres_solver.cpp:256]: :+1:

**Comment by ivonaj** [comment]
Process function for correcting the initial node looks good, and you corrected the solver issue :sunglasses: 
Go for it :rocket:

**Pull request merged**